packages/dev/core/src/AudioV2/abstractAudio/components/spatialAudioAttacherComponent.ts

```diff
@@ -0,0 +1,167 @@
+     * not attached to a transform node.
+     */
+    public get attachedTransformNode(): Nullable<TransformNode> {
+        return this._attacher?.getClassName() === _SpatialAudioAttachedEntity.TRANSFORM_NODE ? (this._attachedEntity as TransformNode) : null;
+    }
+
+    public set attachedTransformNode(value: Nullable<TransformNode>) {
+        if (this.attachedTransformNode === value) {
+            return;
+        }
+
+        this._isReadyPromise = this._resetAttachedEntity(value, _SpatialAudioAttachedEntity.TRANSFORM_NODE);
+    }
+
+    /**
+     * Returns `true` if the audio listener or source is attached to an entity; otherwise returns `false`.
+     */
+    public get isAttached(): boolean {
+        return this._attacher !== null;
+    }
+
+    /**
+     * Returns `true` if the audio listener or source is attached to an entity's position; otherwise returns `false`.
+     */
+    public get isAttachedToPosition(): boolean {
+        return this._attacher !== null && (this.attachmentType & SpatialAudioAttachmentType.POSITION) === SpatialAudioAttachmentType.POSITION;
+    }
+
+    /**
+     * Returns `true` if the audio listener or source is attached to an entity's rotation; otherwise returns `false`.
+     */
+    public get isAttachedToRotation(): boolean {
+        return this._attacher !== null && (this.attachmentType & SpatialAudioAttachmentType.ROTATION) === SpatialAudioAttachmentType.ROTATION;
+    }
+
+    /**
+     * A promise that resolves when the attacher is ready.
+     */
+    public get isReadyPromise(): Promise<void> {
+        return this._isReadyPromise ?? Promise.resolve();
+    }
+
+    /**
+     * Detaches the attached entity.
+     */
+    public detach() {
+        this._attacher?.dispose();
+        this._attacher = null;
+    }
+
+    /**
+     * Releases associated resources.
+     */
+    public dispose() {
+        this.detach();
+    }
+
+    /**
+     * Updates the audio listener or source.
+     */
+    public update() {
+        this._attacher?.update(true);
+    }
+
+    private _createAttacher(attacherClassName: string): Nullable<Promise<_AbstractSpatialAudioAttacher>> {
+        switch (attacherClassName) {
+            case _SpatialAudioAttachedEntity.CAMERA:
+                return this._attachedEntity ? _CreateSpatialAudioCameraAttacherAsync(this) : null;
+            case _SpatialAudioAttachedEntity.MESH:
+                return this._attachedEntity ? _CreateSpatialAudioMeshAttacherAsync(this) : null;
+            case _SpatialAudioAttachedEntity.TRANSFORM_NODE:
+                return this._attachedEntity ? _CreateSpatialAudioTransformNodeAttacherAsync(this) : null;
+        }
+        return null;
+    }
+
+    private async _resetAttachedEntity(entity: Nullable<AbstractMesh | Camera | TransformNode>, attacherClassName: string): Promise<void> {
+        this.detach();
+
+        this._attachedEntity = entity;
+        this._attacher = await this._createAttacher(attacherClassName);
```

Comment:
this is the only reason the entire API is async. I assume it is done to dynamic-load the camera/mesh/transformnode attacher? 
These classes are quite small, and will be smaller with minification, maybe it will be good to have them as a dependency of this abstract class to avoid the async setOptions function.

Reply:
Ya, making this synchronous makes it possible to remove `async` from the `setOptions` functions, too.